src/datasources/asset/AssetDataSource.ts

```diff
@@ -0,0 +1,325 @@
+          } else {
+            requestBody.assetFilter = `${AssetFilterProperties.LocationMinionId} = "${arrayOfEntityIds[index]}" and ${AssetFilterProperties.IsSystemController} = true`;
+          }
+          const response: AssetUtilizationHistoryResponse = await this.post<AssetUtilizationHistoryResponse>(this.baseUrl + '/query-asset-utilization-history', requestBody);
+          requestCount++;
+          continuationToken = response.continuationToken || '';
+          data = data.concat(response.assetUtilizations);
+        } catch (error) {
+          throw new Error(`An error occurred while retrieving utilization history: ${error}`);
+        }
+      } while (continuationToken && (maxDataPoints && requestCount < maxDataPoints));
+
+      // let utilization_data: { id: string, datetimes: Date, values: number[] };
+      let dataWithoutOverlaps: Array<Interval<number>> = [];
+      if (data.length) {
+        const extractedTimestamps = extractTimestampsFromData(data);
+        // Fill missing endTimestamps
+        const patchedData = patchMissingEndTimestamps(extractedTimestamps);
+        // Removes data outside the grafana 'from' and 'to' range from an array
+        const filteredData = filterDataByTimeRange(patchedData, from, to);
+        // Merge overlapping utilizations
+        dataWithoutOverlaps = mergeOverlappingIntervals(filteredData);
+      }
+      // Divide given time range to peak/non-peak intervals
+      const businessIntervals = divideTimeRangeToBusinessIntervals(new Date(from), new Date(to), workingHoursPolicy, peakDays, timeFrequency);
+      // Group raw utilization data by intervals
+      const overlaps = groupDataByIntervals(businessIntervals, dataWithoutOverlaps, peakDays, isPeak);
+      // Calculate utilization percentage by intervals
+      const utilization = calculateUtilization(overlaps);
+      // patch zero points to data for better visualisation
+      const patchedUtilization = patchZeroPoints(utilization, new Date(from), new Date(to), timeFrequency);
+
+      let utilization_data: { id: string, datetimes: number[], values: number[] } =
+        {
+          id: arrayOfEntityIds[index],
+          datetimes: patchedUtilization.map(v => dateTime(v.day).valueOf()),
+          values: patchedUtilization.map(v => v.utilization)
+        }
+      resultArray.push(utilization_data);
+    }
+
+    return resultArray;
+  }
+
+  async getServicePolicy(): Promise<ServicePolicyModel> {
+    return await this.get<ServicePolicyModel>(this.baseUrl + '/policy');
+  }
+
+  async queryAssets(filter = '', take = -1): Promise<AssetModel[]> {
+    let data = { filter, take };
+    try {
+      let response = await this.post<AssetsResponse>(this.baseUrl + '/query-assets', data);
+      return response.assets;
+    } catch (error) {
+      throw new Error(`An error occurred while querying assets: ${error}`);
+    }
+  }
+
+  async querySystems(filter = '', projection = defaultProjection): Promise<SystemMetadata[]> {
```

Comment:
This request is duplicated in the systems data source: https://github.com/ni/systemlink-grafana-plugins/blob/e20476f67532b8fd57628d7311551931921a821a/src/datasources/system/SystemDataSource.ts#L64

Can you single source it in `DataSourceBase` like we did with `getWorkspaces`?